src/shared_types.rs

```diff
@@ -0,0 +1,283 @@
+use serde::de;
+use serde::de::Deserializer;
+use serde::ser;
+use serde::ser::{SerializeStruct, Serializer};
+use serde::{Deserialize, Serialize};
+
+#[cfg(feature = "druid")]
+use druid::Data;
+
+/// Identifiers are optional attributes of several objects in the UFO.
+/// These identifiers are required to be unique within certain contexts
+/// as defined on a per object basis throughout this specification.
+/// Identifiers are specified as a string between one and 100 characters long.
+/// All characters must be in the printable ASCII range, 0x20 to 0x7E.
+#[derive(Debug, Clone, PartialEq)]
+pub struct Identifier(pub(crate) String);
+
+impl Identifier {
+    fn is_valid(&self) -> bool {
+        if self.0.len() > 100 {
```

Comment:
I would probably write this as,

```rust
fn is_valid(&self) -> bool {
    self.0.len() <= 100
        && self.0.as_bytes().all(|b| b >= 0x20 && b < 0x7F)
}

Reply:
Agreed, but boy do I find Rust's terseness alien coming from Python.